src/lib.rs

```diff
@@ -150,129 +147,75 @@ fn generate_and_load_analysis(config: &Config) -> Result<()> {
-                ids.push(id);
-                def.clone()
-            }).unwrap();
-
-            let child_defs: Vec<analysis::Def> = ids.into_par_iter()
-                .map(|id: analysis::Id| recur(&id, host))
-                .reduce(|| Vec::new(), |mut a: Vec<analysis::Def>,
-                 b: Vec<analysis::Def>| {
-                    a.extend(b);
-                    a
-                });
-            defs.extend(child_defs);
-            defs
-        }
-
-        doc_data.metadata = recur(&root_id, host)
-            .into_iter()
-            .map(|def| {
-                Metadata {
-                    kind: def.kind,
-                    qualified_name: def.qualname,
-                    name: def.name,
-                    docs: def.docs,
-                }
-            })
-            .collect();
-
-        Ok(doc_data)
+        // Using the item's metadata we create a new `Document` type to be put in the eventual
+        // serialized JSON.
+        included.push(
+            Document::new()
+                .ty(ty.clone())
+                .id(def.qualname.clone())
+                .attributes(String::from("name"), def.name)
+                .attributes(String::from("docs"), def.docs),
+        );
+
+        let item_relationships = relationships.entry(relations_key).or_insert_with(
+            Default::default,
+        );
+        item_relationships.push(Data::new().ty(ty).id(def.qualname));
     }
 
-    /// Serialize the data structure into a valid `JSON API` String for output later.
-    pub fn to_json(&self) -> Result<String> {
-        // Set up the values we'll later push into the to `Documentation` struct to be serialized
-        let mut included: Vec<Document> = Vec::new();
-        let mut relationships: HashMap<&str, Vec<Data>> = HashMap::with_capacity(METADATA_SIZE);
-
-        // Check each item in the metadata and add it to be serialized based off it's type
-        for item in &self.metadata {
-            let (ty, relations_key) = match item.kind {
-                DefKind::Mod => ("module", "modules"),
-                DefKind::Struct => ("struct", "structs"),
-                _ => continue,
-            };
-
-            // Using the item's metadata we create a new `Document` type to be put in the eventual
-            // serialized JSON.
-            included.push(
-                Document::new()
-                    .ty(ty)
-                    .id(&item.qualified_name)
-                    .attributes("name", &item.name)
-                    .attributes("docs", &item.docs),
-            );
-
-            let item_relationships = relationships.entry(relations_key).or_insert_with(
-                Default::default,
-            );
-            item_relationships.push(Data::new().ty(ty).id(&item.qualified_name));
-        }
-
-        let len = self.name.len();
+    let mut data_document = Document::new()
+        .ty(String::from("crate"))
+        .id(crate_name.to_string())
```

Comment:
Why is this `to_string()` instead of `String::from()`? On a side note why not just use `into()`? For more explicitness?